sdk/core/Azure.Core.TestFramework/src/VersionString.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.TestFramework
+{
+    internal readonly struct VersionString : IComparable
+    {
+        private bool IsGa { get; }
+        private string PreviewString { get; }
+        private string RawVersion { get; }
+        private DateTime VersionDate { get; }
+
+        public VersionString(string version)
+        {
+            Argument.AssertNotNull(version, nameof(version));
+
+            RawVersion = version;
+            ReadOnlySpan<char> chars = version.AsSpan();
+            IsGa = true;
+            int index = -1;
+            PreviewString = null;
+            if (chars.Length > 10)
```

Comment:
Why 10?

Reply:
I think that is the standard length for a version string minus any preview suffixes. `nnnn-nn-nn`

Reply:
Yes, that is what I assumed, but I am hinting at the need for a constant or a comment.